### ISSUE TITLE
Xvfb is breaking wkhtmltopdf output

### ISSUE THREAD

**Issue by navi2000** [wkhtmltopdf/wkhtmltopdf#2361]
This is ok:

wkhtmltopdf2 --use-xserver --no-stop-slow-scripts -B 0 -L 0 -R 0 -T 0 cover "cover.html" "test.html" out.pdf

but this fails (adds some space in the cover right and bottom margins):

xvfb-run wkhtmltopdf2 --use-xserver --no-stop-slow-scripts -B 0 -L 0 -R 0 -T 0 cover "cover.html" "test.html" out.pdf

even if I use parameters like --server-args="-screen 0, 1280x1024x24". No success.

My test.html is:

```
 <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"> 
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="es" lang="es">
 <head>
     <title>Test</title>
 </head>

 <body style="margin:0;padding:0">
     <p>Hello world</p>   
 </body> 
 </html>
```

(simple HTML "Hello world")

And my cover.html is:

```
 <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="es" lang="es">
   <head>
     <title>Cover Test</title>
   </head>

   <body style="margin:0;padding:0">
     <div style="background-color:red; width:1000px; height:1414px">
     </div>
   </body>
 </html>
```

A simple big red box. If I use these html files without xvfb I get a perfect red box covering all the space of the cover (without any spaces in the margins). If I use xvfb I get a red box with a right and a bottom white margin.

I'm using latest version (qt patched 0.12.2.1)

**Comment by Petoj87** [comment]
i am no linux guy by any means but have you tried xdummy?

**Comment by navi2000** [comment]
AFAIK xdummy (part of Xpra) uses xvfb, so are equivalents. Also, I've never seen any script using Xpra from webservices (that's the main use case to make use of xvfb)

**Comment by strk** [comment]
why does wkhtmltopdf needs to connect to an X server anyway ? Can it be asked NOT to ?

**Comment by navi2000** [comment]
Because of this: https://coderwall.com/p/tog9eq/using-wkhtmltopdf-and-an-xvfb-daemon-to-render-html-to-pdf
My particular use case is I am converting HTML to PDF in a server to serve a document composed in real time using user input

**Comment by strk** [comment]
That article does not respond to the question: WHY is an X server needed at all ?

**Comment by navi2000** [comment]
That article is why I'm using this, not why I need it. I'm not a Linux expert, but when I've tryed to use wkhtmltopdf without xvfb I haven't had any result (BTW my last test was when I first posted this, around 2015). When I asked about this, some people answered that I need an environment and pointed me to some articles like this.

So, maybe it's not necessary anymore, BUT some years ago without using the  --use-xserver and xvfb, it was not possible to render a PDF properly. And in any case, it's a bug if the rendered result is different from expected, isn't it?
